src/e_compression.c

```diff
@@ -787,7 +787,10 @@ int32_t e_comp(int32_t *in,
       for (k = 0; k < didsamp; k++) {
         *out++ = htonl(*(uint32_t *)in++);
       }
-      *outbytes += bufbytes;
+      uint16_t block_bytes = (uint16_t)(sizeof(uint32_t)*(out - pout));
```

Comment:
I've been trying really hard to not touch the C code, as it's been working for us for decades (and I don't understand it 😄  ).  Can you explain what this is?

Reply:
I’m with you, if the C code has been stable for years, it’s safer to leave it alone. I initially patched the header logic inside the C file because it was the most direct (and fastest) fix, but I also prototyped with a pure‑Python workaround while I built up my understanding. I’ll push that as a separate PR (#7).

The Python wrapper now splits the data into full blocks and a final tail block, passing the existing EC_SHORT_END flag for the tail. That yields the same “correct header size” behavior as the C change in this PR, with zero changes to the legacy source, just a small performance trade‑off on the Python side.